plugins/modules/digital_ocean_domain.py

```diff
@@ -208,10 +226,7 @@ def main():
         ),
     )
 
-    try:
-        core(module)
-    except Exception as e:
-        module.fail_json(msg=to_native(e), exception=traceback.format_exc())
+    run(module)
```

Comment:
I'm curious about this change; catching `Exception` is certainly overly broad and should be avoided, but without it, how is it handled? Does the framework transparently handle the exception and similarly issue a `fail_json` or similar?

Reply:
Seems like it ... I forced an `Exception` and got this:

```
TASK [Create a domain] **********************************************************************************************************************************
Tuesday 13 April 2021  18:25:39 -0400 (0:00:00.068)       0:00:00.068 ********* 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Exception: foo
[WARNING]: Platform darwin on host localhost is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python
interpreter could change the meaning of that path. See https://docs.ansible.com/ansible/2.10/reference_appendices/interpreter_discovery.html for more
information.
fatal: [localhost]: FAILED! => changed=false 
  ansible_facts:
    discovered_interpreter_python: /usr/bin/python
  module_stderr: |-
    Traceback (most recent call last):
      File "/Users/mark/.ansible/tmp/ansible-tmp-1618352739.748239-91987-140231147269929/AnsiballZ_digital_ocean_domain.py", line 102, in <module>
        _ansiballz_main()
      File "/Users/mark/.ansible/tmp/ansible-tmp ...
```

...it's neatly handling it as a module failure and dumping the traceback to `module_stderr`.

Reply:
Well, things are really simple. Ansible packages the module into a standalone application and runs it. If everything goes as planned, the application exits with a status 0 and prints serialized results to stdout.

If things go haywire (like in the case of non-handled exception), the program exits with a non-zero status and Ansible simply captures the stdout and stderr of the process.

So, the rule of thumb our team has: only catch errors that we expect and can provide a better diagnostic message to the user. If we can only return stacktrace, we have a bug in our code and the module should fail in an ugly way so we can fix it. For example, https://github.com/ansible-collections/servicenow.itsm/blob/main/docs/module_template.md is how we usually structure our code.

Reply:
> Well, things are really simple. Ansible packages the module into a standalone application and runs it. If everything goes as planned, the application exits with a status 0 and prints serialized results to stdout.
> 
> If things go haywire (like in the case of non-handled exception), the program exits with a non-zero status and Ansible simply captures the stdout and stderr of the process.
> 
> So, the rule of thumb our team has: only catch errors that we expect and can provide a better diagnostic message to the user. If we can only return stacktrace, we have a bug in our code and the module should fail in an ugly way so we can fix it. For example, https://github.com/ansible-collections/servicenow.itsm/blob/main/docs/module_template.md is how we usually structure our code.

Thanks for sharing this @tadeboro, I ⭐ 'd the template, thank you. Over time, I'd like to scour this codebase with an emphasis on consistency. I think we're going to lots of refactoring for 2.0 -- I've not liked `digital_ocean` (versus `digitalocean` for quite some time). Also, I don't really like the redundancy in the module names, e.g., I'd much rather see `community.digitalocean.droplet` versus `community.digitalocean.digital_ocean_droplet`.